cumulus/steps/ec2/listener_rule.py

```diff
@@ -0,0 +1,67 @@
+from troposphere import elasticloadbalancingv2 as alb, route53
+from troposphere import (
+    Ref, Join, ec2)
+from cumulus.chain import step
+from cumulus.steps.ec2 import META_TARGET_GROUP_NAME
+
+
+class ListenerRule(step.Step):
+
+    def __init__(self,
+                 base_domain_name,
+                 alb_listener_rule,
+                 priority,
+                 path_pattern=None,
+                 host_pattern=None,
+                 ):
+
+        step.Step.__init__(self)
+
+        self.path_pattern = path_pattern
+        self.host_pattern = host_pattern
+        self.priority = priority
+        self.base_domain_name = base_domain_name
+        self.alb_listener_rule = alb_listener_rule
+
+    def handle(self, chain_context):
+        ALB_LISTENER_RULE = "ListenerRule"
+
+        template = chain_context.template
+
+        if not (self.path_pattern or self.host_pattern):
+            raise RuntimeError("with_listener_rule() requires one of: path_pattern, host_pattern")
+
+        routing_condition = None
+        if self.path_pattern:
+            routing_condition = alb.Condition(
+                Field="path-pattern",
+                Values=[self.path_pattern],
+            )
+        elif self.host_pattern:
+            routing_condition = alb.Condition(
+                Field="host-header",
+                Values=[
+                    Ref("stackname"),
+                    "-",
+                    Ref("namespace"),
```

Comment:
namespace, and stackname again.   Are these bleeding in from stacker?   Namespace is a stacker concept.. although the code that this was brought in from.. (the builder) didn't have a working host_pattern.   So maybe this could be tackled separately.. as I'm guessing this isn't tested in this PR.